basic-tracer/src/Cardano/BM/Tracer/Class.lhs

```diff
@@ -22,7 +24,7 @@ annotate) such observations with additional information from their
 execution context.
 
 \begin{code}
-newtype Tracer m s = Tracer { tracing :: Op (m ()) s }
+newtype Tracer m s = Tracer { trace :: s -> m () }
```

Comment:
If this patch is about simplification, then here's an alternative argument: use `Op (m ()) s` as before, and derive the `Contravariant` instance. Many would argue that is simpler than what we have here.

Reply:
the renaming of `tracing` to `trace` lead to a number of conflicts. I would prefer to keep `traciing`.

Reply:
Or keep `traceWith` if that's the name we like at the use sites, but it can be the record field function. Or `tracing` as the function we use at the use sites. The point is, having eliminated `Op` the record field function _is_ the function we can use at use sites, so lets pick it appropriately.

I think eliminating `Op` is a win for the use sites. Sure it allows deriving `Contravariant` here, but that saves us a couple lines at the definition site whereas all use sites have to pull in `Op` and use it. It adds nothing.

Reply:
The is first-degree bikeshedding. I'm amazed that [what I committed almost a year ago](https://github.com/input-output-hk/cardano-sl/commit/45e8d9094ef7545cc470ba2b4b6023cf4a73580b#diff-275208aecbc057b515b82a5b8c7acc19) is all of a sudden too complicated.

Reply:
I would have been happy with that. Eliminating the `Op` thing is minor, but a minor improvement. Note that there you provide `trace` as the function to use at the use site. Apparently we now think that's too good a name to steal. :man_shrugging:

Reply:
@CodiePP If you're really not happy with the record field function being `trace` then take the suggestion from @avieth and use
```
newtype Tracer m s = Tracer { runTracer :: s -> m () } 

trace :: Tracer m s -> s -> m ()
trace = runTracer
```
That way, when you import `Tracer(..)` in the other internal modules, you can do so without bringing `trace` into scope, but use sites of tracing can still (if they like) import and use `trace` or `traceWith`.

@avieth would that work for you too? You complained:
> I'd like to be able to name my trace parameter trace, but can't do that if the record field trace is in scope.

Reply:
Very good points!
I think I go for `runTracer` field and the function being named `traceWith` to avoid name clashes. ..